hivemind/optim/collaborative.py

```diff
@@ -177,26 +184,32 @@ def step(self, batch_size: Optional[int] = None, **kwargs):
         with self.performance_ema.pause(), self.lock_collaboration_state:
             # divide accumulators by local steps to recover the true average grad w.r.t. local_samples_accumulated
             self.apply_accumulated_grads_(scale_by=1. / self.local_steps_accumulated)
+            current_step, group_info = self.averager.local_step, None
 
             if self.collaboration_state.num_peers > 1:
                 mean_samples_per_worker = self.target_batch_size / self.collaboration_state.num_peers
                 weight = self.local_samples_accumulated / mean_samples_per_worker
-                output = self.averager.step(weight=weight, timeout=self.averaging_timeout, **kwargs)
+                try:
+                    group_info = self.averager.step(weight=weight, timeout=self.averaging_timeout, **kwargs)
+                    if group_info:
+                        logger.log(self.status_loglevel, f"Averaged tensors successfully with {len(group_info)} peers")
+                except Exception as e:
+                    logger.log(self.status_loglevel, f"Skipped averaging: averaging round failed with {e}.")
+
             else:
                 logger.log(self.status_loglevel, f"Skipped averaging: collaboration consists of "
                                                  f"{self.collaboration_state.num_peers} peer(s).")
-                output = None
-                self.averager.local_step += 1
 
             self.opt.step()
             self.reset_accumulated_grads_()
             self.local_samples_accumulated = self.local_steps_accumulated = 0
-            self.collaboration_state.register_step()
+            self.collaboration_state.register_step(current_step + 1)
+            self.averager.local_step = current_step + 1
```

Comment:
The step number of collaboration_state and averager might differ here (because of max)

Reply:
yep, that is the intended behavior: local peer can be behind the collaboration. If it is too far behind, it will attempt to load_state_from_peers

Reply:
Got it, thanks!